kong/router/atc.lua

```diff
@@ -553,14 +625,29 @@ function _M:exec(ctx)
     headers_key = get_headers_key(headers)
   end
 
+  local queries, queries_key
+  if self.match_queries then
+    local err
+    queries, err = get_uri_args()
+    if err == "truncated" then
+      local lua_max_req_queries = kong and kong.configuration and kong.configuration.lua_max_req_queries or 100
```

Comment:
Should we add a new option `lua_max_req_queries` in kong.conf? like `lua_max_req_headers`.

Reply:
Let's just use hard coded 100 for now, it should be fine for 99% of users. Log the error when response is truncated.

Reply:
done.